src/main/java/org/broadinstitute/hellbender/engine/ReadsDataSource.java

```diff
@@ -39,9 +41,10 @@
     private final Map<SamReader, CloseableIterator<SAMRecord>> readers;
 
     /**
-     * Hang onto the input files so that we can print useful errors about them
+     * Hang onto the input files so that we can print useful errors about them.
+     * The map is bi directional so that we can look up the reader for a given file.
      */
-    private final Map<SamReader, File> backingFiles;
+    private final BidiMap<SamReader, File> backingFiles;
```

Comment:
I don't see why this needs to be bidirectional.  `getKey(File)` is used but `getValue(SamReader)` is not, so it seems like we could equally well switch the keys and values to get a `Map<File, SamReader>`.